### ISSUE TITLE
cmake help

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1899]
I want to use cmake but not use the bundled mpark, but the installed one.
The [docs](https://bout-dev.readthedocs.io/en/latest/user_docs/installing.html#cmake) say I should run `cmake .. -LH`, but that does run:
```
Cloning into '/home/dave/soft/BOUT-dev/realy/externalpackages/mpark.variant'...
```
Which is exactly what I do not want.
Is there a way around this?

I tried again, disabling internet, and then I got:
```
$ cmake .. -LH
-- The CXX compiler identification is GNU 9.2.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MPI_CXX: /usr/lib64/mpich/lib/libmpicxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Submodule update
Cloning into '/BOUT-dev/externalpackages/git-archive-all.sh'...
fatal: unable to access 'https://github.com/meitar/git-archive-all.sh/': Failed to connect to github.com port 443: Connection refused
fatal: clone of 'https://github.com/meitar/git-archive-all.sh/' into submodule path '/BOUT-dev/externalpackages/git-archive-all.sh' failed
Failed to clone 'externalpackages/git-archive-all.sh'. Retry scheduled
Cloning into '/BOUT-dev/externalpackages/googletest'...
fatal: unable to access 'https://github.com/google/googletest.git/': Failed to connect to github.com port 443: Connection refused
fatal: clone of 'https://github.com/google/googletest.git' into submodule path '/BOUT-dev/externalpackages/googletest' failed
Failed to clone 'externalpackages/googletest'. Retry scheduled
Cloning into '/BOUT-dev/externalpackages/mpark.variant'...
fatal: unable to access 'https://github.com/mpark/variant.git/': Failed to connect to github.com port 443: Connection refused
fatal: clone of 'https://github.com/mpark/variant.git' into submodule path '/BOUT-dev/externalpackages/mpark.variant' failed
Failed to clone 'externalpackages/mpark.variant'. Retry scheduled
Cloning into '/BOUT-dev/externalpackages/git-archive-all.sh'...
fatal: unable to access 'https://github.com/meitar/git-archive-all.sh/': Failed to connect to github.com port 443: Connection refused
fatal: clone of 'https://github.com/meitar/git-archive-all.sh/' into submodule path '/BOUT-dev/externalpackages/git-archive-all.sh' failed
Failed to clone 'externalpackages/git-archive-all.sh' a second time, aborting
CMake Error at CMakeLists.txt:32 (message):
  git submodule update --init failed with 1, please checkout submodules


-- Configuring incomplete, errors occurred!
See also "/BOUT-dev/build/CMakeFiles/CMakeOutput.log".
-- Cache values
// Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel ...
CMAKE_BUILD_TYPE:STRING=

// Install path prefix, prepended onto install directories.
CMAKE_INSTALL_PREFIX:PATH=/usr/local

// Check submodules during build
GIT_SUBMODULE:BOOL=ON
```

Thus I tried:
```
$ cmake .. -LH -DGIT_SUBMODULE=OFF
-- The CXX compiler identification is GNU 9.2.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MPI_CXX: /usr/lib64/mpich/lib/libmpicxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
CMake Error at CMakeLists.txt:38 (message):
  The submodules were not downloaded! GIT_SUBMODULE was turned off or failed.
  Please update submodules and try again.


-- Configuring incomplete, errors occurred!
See also "/BOUT-dev/build/CMakeFiles/CMakeOutput.log".
-- Cache values
// Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel ...
CMAKE_BUILD_TYPE:STRING=

// Install path prefix, prepended onto install directories.
CMAKE_INSTALL_PREFIX:PATH=/usr/local

// Check submodules during build
GIT_SUBMODULE:BOOL=OFF
```

Which isn't working either.

Is there anyway to get cmake working without using the bundled libraries?

Also it seems the default check level is 3 with cmake, while it is 2 for configure. Should I file an extra bug for this?

**Comment by ZedThree** [comment]
Oh yeah, looks like that's not currently possible. I'll fix it now and update the docs so it's clearer.

**Comment by ZedThree** [comment]
Fixed in #1900
